roles/nos-notification-service/README.md

```diff
@@ -0,0 +1,26 @@
+# Nos Notification Service
+
+This role sets up the [nos notification
+service](https://github.com/planetary-social/nos-notification-service-go/) on an
+ubuntu server.
+
+It is assumed that server has already had the common and harden roles applied to it.  
```

Comment:
Is there a reason these dependencies aren't explicit in `roles/nos-notification-service/meta/main.yml` along with the other dependent roles listed there?

Reply:
In short, the state of the target server for those first two roles is different from all the roles following.  At the start it is a server with only the root user and password login.  These first two roles add the admin user and remove the ability to login as root or login with a password.  

The roles that follow are run as a different ansible_user than these first two (from root to admin), which affects the ownership and permissions of the files that get added to the servers.

I researched ways to switch the ansible_user mid playbook to manage something like this, but couldn't find a satisfactory way.  There may be a better way to handle this though.

Reply:
(I should add that the roles added in /meta/main.yml as dependencies means that the playbook will run those roles first, if they haven't already been ran.  So I didn't want to add common and harden, as the playbook would try to run them as the admin user, which would be incorrect).

Reply:
ah makes sense, thanks for explaining that. So what are the steps for hardening a new server? Is there a playbook that you would run on a new machine before i.e. notifications-playbook.yml?

Reply:
There is a minimal playbook you would run, that is basically:
```yaml
- name: Setup and harden server
  hosts: all
  roles:
    - harden
  ```
  
  I opened a ticket (#39) to add this playbook to the repo, so it is not internal knowledge.